downstream.go

```diff
@@ -298,7 +299,12 @@ func (dc *downstreamConn) marshalMessage(msg *irc.Message, net *network) *irc.Me
 	case "NICK":
 		// Nick change for another user
 		msg.Params[0] = dc.marshalEntity(net, msg.Params[0])
-	case "JOIN", "PART":
+	case "JOIN":
+		msg.Params[0] = dc.marshalEntity(net, msg.Params[0])
+		if !dc.caps["extended-join"] {
```

Comment:
I think this should be moved to `SendMessage`.

Reply:
Updated